src/Compilers/Core/CommandLine/BuildClient.cs

```diff
@@ -93,13 +93,14 @@ internal int RunCompilation(IEnumerable<string> originalArguments, BuildPaths bu
                         return HandleResponse(buildResponse, parsedArgs, buildPaths);
                     }
                 }
-                catch
+                catch (OperationCanceledException)
                 {
-                    // It's okay, and expected, for the server compilation to fail.  In that case just fall 
-                    // back to normal compilation. 
+                    return CommonCompiler.Succeeded;
```

Comment:
It's a virtual method, it can throw anything. This is why I had the catch all originally. 

Do we have a bug tracking the fix for the retune value here?